### PR TITLE
Updated doc for consumers authority

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
**Changes proposed in this request**
Updated documentation for issue 2929.

**Testing**
NA

**Performance impact**
None

**Documentation**
- [x] All relevant documentation is updated.

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs [hunk 1]
```diff
@@ -461,7 +461,9 @@ public static IAuthorityValidator CreateAuthorityValidator(AuthorityInfo authori
             ///
             /// - if the authority is not defined at the application level and the request level is not AAD, use the request authority
             /// - if the authority is defined at app level, and the request level authority of is of different type, throw an exception
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs:463]: needs some rewording here?

File: src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs [hunk 2]
```diff
@@ -461,7 +461,9 @@ public static IAuthorityValidator CreateAuthorityValidator(AuthorityInfo authori
...
-            ///
+            /// 
+            /// - if the intended authority is consumers, please define it at the app level and not at the request level.
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs:465]: has this been tested for the scenario mentioned in the bug?

Comment (SameerK-MSFT) [reply src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs:465]: Good point. Tests pushed.

File: src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs (hunk 1 above)
Comment (pmaytak) [reply src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs:463]: Reword how? We throw if app and request authorities are of different types (AAD, B2C, ADFS) or have different hosts/clouds.

Comment (SameerK-MSFT) [reply src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs:463]: I think he means grammatical error: remove "of" after the authority.

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs [hunk 3]
```diff
@@ -916,6 +916,62 @@ public async Task GetAuthorizationRequestUrl_NoHint_NoCcsRoutingHint_TestAsync()
...
+                Uri authorizationRequestUrl = confidentialClientApplication
+                    .GetAuthorizationRequestUrl(new List<string> { "" })
+                    .ExecuteAsync()
+                    .ConfigureAwait(false)
+                    .GetAwaiter()
+                    .GetResult();
+
+                Assert.IsTrue(authorizationRequestUrl.Segments[1].StartsWith(Tenant));
+            }
+        }
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs:945]: The idea is that if request changes authority to common/org/consumer, config takes precendence.

```suggestion
        public void GetAuthorizationRequestUrl_WithConsumerInCreate_ReturnsConsumers()
        {
            using (var httpManager = new MockHttpManager())
            {
                ConfidentialClientApplicationOptions applicationOptions;
                applicationOptions = new ConfidentialClientApplicationOptions();
                applicationOptions.ClientId = "fakeId";
                applicationOptions.RedirectUri = "https://example.com";
                applicationOptions.ClientSecret = "rwerewrwe";

                var confidentialClientApplicationBuilder = ConfidentialClientApplicationBuilder
                                                                    .CreateWithApplicationOptions(applicationOptions)
                var confidentialClientApplication = confidentialClientApplicationBuilder.Build();

                Uri authorizationRequestUrl = confidentialClientApplication
                    .GetAuthorizationRequestUrl(new List<string> { "" })
                    .WithAuthority(AzureCloudInstance.AzurePublic, Constants.ConsumerTenant);
                    .ExecuteAsync()
                    .ConfigureAwait(false)
                    .GetAwaiter()
                    .GetResult();

                Assert.IsTrue(authorizationRequestUrl.Segments[1].StartsWith(Constants.CommonTenant));
            }
        }
```

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs [hunk 4]
```diff
@@ -916,6 +916,62 @@ public async Task GetAuthorizationRequestUrl_NoHint_NoCcsRoutingHint_TestAsync()
...
+
+        [TestMethod]
+        public void GetAuthorizationRequestUrl_WithConsumerAudienceInCreate_ReturnsConsumers()
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs:948]: This is not needed.

**Review by pmaytak — APPROVED** [review]

**Review by neha-bhargava — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs:465]: Yeah, hide it. We want to allow only `WithTenant` at request level. 
Comments should go on `WithTenant`

File: tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs (hunk 4 above)
Comment (SameerK-MSFT) [reply tests/Microsoft.Identity.Test.Unit/PublicApiTests/ConfidentialClientApplicationTests.cs]: Removed

**Pull request merged**